packages/components/Icon/src/Icon.types.ts

```diff
@@ -0,0 +1,35 @@
+import { SvgProps } from 'react-native-svg';
+export const iconName = 'Icon';
+
+export interface IFontIconProps
+{
+  fontFamily?: string;
+  fontSrcFile?: string;
+  codepoint: number;
+  fontSize?: number;
+  color?: string;
+}
+
+export interface ISvgIconProps
+{
+  uri?: string;
+  src?: React.FC<SvgProps>;
+  viewBox?: string;
+  color?: string;
+}
+
+export interface IRasterImageIconProps
+{
+  src: object;
+}
+
+export interface IIconProps
+{
+  ariaLabel?: string;
+  fontSource?: IFontIconProps;
+  svgSource?: ISvgIconProps;
+  rasterImageSource?: IRasterImageIconProps;
+  width?: number;
+  height?: number;
+  color?: string;
```

Comment:
Same with color, I would expect that if someone sets this value it will be moved to the right field in the sub-objects. I wouldn't expect users to have to set those sub-objects.

Reply:
Oh yes color shouldn't be in IIconProps but only in IFontIconProps and ISvgIconProps since only those are recolorable and RasterIcons are not.